contracts/ZKPVerifier.sol

```diff
@@ -7,7 +7,15 @@ import "./lib/GenesisUtils.sol";
 import "./interfaces/ICircuitValidator.sol";
 import "./interfaces/IZKPVerifier.sol";
 
+interface ISpongePoseidon {
+    function hash(uint256[] calldata values) external view returns (uint256);
+
+    function hash4(uint256[4] calldata values) external view returns (uint256);
+}
+
 contract ZKPVerifier is IZKPVerifier, Ownable {
+    ISpongePoseidon public poseidon;
```

Comment:
Any strong reason to keep it public? Solidity compiler will generate the getters code and you need to implement `setSpongePoseidon` method, general complexity increases, attack surface increases? Was it discussed with pros&cons?
If no, you can do it private, and rename to `_poseidon` and remove the setter method.

Reply:
No reason, will change